Update bulk-index to support bulk-indexing all

nonword opened the pull request:
The hasMarc and nyplSource filters are now optional

charmingduchess commented on scripts/bulk-index.js:
```diff
@@ -613,7 +623,7 @@ const run = async () => {
     argv.bibId ||
     argv.itemId ||
-    (argv.type && argv.hasMarc)
+    (argv.type)
```
Does this leave the script open to an update on all of a given type? I feel like we should encourage a practice of narrowing down slightly the scope... maybe a check like `argv.type && (argv.hasMarc || argv.nyplSource)`

nonword replied:
Will implement

nonword commented:
I updated the arg validation to require either nyplSource or hasMarc (although looks like currently nyplSource defaults to 'sierra-nypl', so it's already effectively required). And I updated the USEAGE note to acknowledge the new useage
